### ISSUE TITLE
remove semi-verified state

### ISSUE THREAD

**Issue by LaurenWags** [brave/brave-browser#15390]
## Description 

Related issues:
https://github.com/brave/brave-browser/issues/15028
https://github.com/brave/brave-browser/issues/13151

Notification to alert user to reaching linking limit has been implemented via above issues, however blocking the connection (for Uphold wallets) has not been added yet.

When this is implemented, any additional wallet providers should be at least spot checked to confirm their existing linkage functionality is still working as designed.

Additional details/information can be obtained from @Miyayes @zenparsing @emerick

**Comment by Miyayes** [comment]
Just checking: will this issue also build this in for Android? If not, we can remove the Android label.

**Comment by zenparsing** [comment]
@Miyayes Yes - the changes will be in the shared rewards engine code.

**Comment by LaurenWags** [comment]
Suggested test plan:
- Run test cases from the `Uphold` section under https://github.com/brave/internal/issues/788
- Upgrades with Uphold account in various states where ever possible (suggestions: Verified, Connected, Pending, Disconnected, etc)

cc @szilardszaloki @zenparsing @Miyayes @brave/legacy_qa

**Comment by LaurenWags** [comment]
Verification in progress using

```
Brave | 1.29.37 Chromium: 92.0.4515.115 (Official Build) nightly (x86_64)
-- | --
Revision | 48cb2f4029b84b003719740a6cf9ca73f374a857-refs/branch-heads/4515_105@{#4}
OS | macOS Version 10.15.7 (Build 19H1217)
```

<details>
<summary>Upgrade - anon/virtual BAT - PASSED</summary>

1. Using current release version, 1.27.109, clean profile, staging env, enable Rewards, claim UGP grant, note balance
2. Close browser
3. "Upgrade" (rename profile to `Brave-Browser-Nightly`) to 1.29.x
4. Confirm balance is still the same post upgrade
5. Confirm you are able to tip

1.27.x | 1.29.x | 1.29.x post tip
----- | ----- | -----
<img width="1193" alt="1 27 x" src="https://user-images.githubusercontent.com/28145373/127548190-6c1868a9-3e68-431a-a6f4-98613e9f4699.png"> | <img width="1185" alt="1 29 x" src="https://user-images.githubusercontent.com/28145373/127548195-e2152776-c485-4897-a8d0-a6127408ceaa.png"> | <img width="1185" alt="post tip" src="https://user-images.githubusercontent.com/28145373/127548198-80c19b6e-b858-4487-9704-6f064c8b7bb0.png">


</details>

**Comment by GeetaSarvadnya** [comment]
Verification is inprogress

```
Brave | 1.30.9 Chromium: 92.0.4515.131 (Official Build) nightly (64-bit)
-- | --
Revision | 6b8d6c56ce21e38a72f7c4becb5abc1fa5134f29-refs/branch-heads/4515@{#1933}
OS | Windows 10 OS Version 2009 (Build 19043.1110)

```
<details>
<summary>Upgrade (NOT_CONNECTED) - anon/virtual BAT - PASSED</summary>

1. Using current release version, 1.27.109, clean profile, staging env, enable Rewards, claim UGP grant, note balance
2. Close browser
3. "Upgrade" (rename profile to `Brave-Browser-Nightly`) to 1.29.x
4. Confirm balance is still the same post upgrade
5. Confirm you are able to tip

1.27.x | 1.29.x | 1.29.x post tip
----- | ----- | -----
![image](https://user-images.githubusercontent.com/38657976/128753997-1bb4afb1-9072-4206-b9d8-f3869d6feb32.png) | ![image](https://user-images.githubusercontent.com/38657976/128754218-a75e0c20-3e6b-46fb-a4e8-ca0faf8a075b.png) | ![image](https://user-images.githubusercontent.com/38657976/128754450-d4890762-035a-4b5a-a976-8b9bb69296eb.png)

**Comment by srirambv** [comment]
Verification passed on Oppo Reno 5 with Android 11 running 1.29.74 x64 build

<details>
<summary>Scenario 1: <code>Upgrade (NOT_CONNECTED) - anon/virtual BAT</code></summary>
<br/>

- Installed current release version, 1.28.106,
- Enabled Rewards and claimed UGP Grant, Verified wallet balance
- Exit browser
- Upgrade to 1.29.74 build
- Verified  balance is retained during upgrade
- Verified able to tip a publisher post upgrade and balance updates accordingly

1.28.x | 1.29.x | Tip sent on 1.29.x
-- | -- | --
![image](https://user-images.githubusercontent.com/17010094/131307421-75647106-4df1-4fb5-a0dd-b591c2419f3b.png) | ![image](https://user-images.githubusercontent.com/17010094/131307438-b92b8181-cd03-461e-b8a8-719ebd85fd6f.png) | ![image](https://user-images.githubusercontent.com/17010094/131307451-7e63a41f-04a3-4f05-becf-0980603bb6b9.png)

</details>

<details>
<summary>Scenario 2: <code>Upgrade (VERIFIED) - Verified KYC'd wallet linked</code></summary>
<br/>

- Installed current release version, 1.28.106,
- Enabled Rewards and connected to a KYC wallet
- Exit browser
- Upgrade to 1.29.74 build
- Verified  wallet is still connected post upgrade
- Verified able to tip a publisher post upgrade and wallet balance updates accordingly

1.28.x | 1.29.x | Tip sent on 1.29.x
-- | -- | --
![image](https://user-images.githubusercontent.com/17010094/131308920-15194b73-542f-47c0-ada9-cfb4e97e5ca6.png) | ![image](https://user-images.githubusercontent.com/17010094/131308941-4399b612-28cf-48d3-a297-f21d5cd2cdf7.png) | ![image](https://user-images.githubusercontent.com/17010094/131308963-c38f16de-4469-45b5-bef0-75f9f896710b.png)

</details>

<details>
<summary>Scenario 3: <code>Upgrade (DISCONNECTED_VERIFIED) - Verified, KYC'd wallet is Disconnected prior to upgrade</code></summary>
<br/>


- Installed current release version, 1.28.106,
- Enabled Rewards and connected to a KYC wallet
- Disconnect the wallet via Uphold, ensured panel shows disconnected message
- Exit browser
- Upgrade to 1.29.74 build
- Verified  wallet is still disconnected post upgrade
- Verified able to re-connect the KYC wallet again post upgrade
- Verified able to tip a publisher post upgrade and wallet balance updates accordingly

1.28.x Disconencted | 1.29.x Disconnected | 1.29.x Re-connected | Tip sent on 1.29.x
-- | -- | -- | --
![image](https://user-images.githubusercontent.com/17010094/131317175-49c8c7d0-a163-42c9-9fd3-4f5bd4cdac04.png) | ![image](https://user-images.githubusercontent.com/17010094/131317240-037c3730-d26d-4e94-a591-581029f9974c.png) | ![image](https://user-images.githubusercontent.com/17010094/131317267-3587c95c-2bb9-45a3-bcb4-66267183c8b7.png) | ![image](https://user-images.githubusercontent.com/17010094/131317283-70e984bd-6691-4499-a495-d4e3cb3c280e.png)

</details>

<details>
<summary>Scenario 4: <code>Upgrade (a SEMI-VERIFIED state) - link 2 KYC'd Uphold accounts prior to upgrade</code></summary>
<br/>

- Installed  current release version, 1.28.106
- Launched staging env and  enable Rewards, link KYC'd Uphold wallet
- Manually disconnect from Uphold (from wallet panel)
- Verified  returned to pre-link state (UI, no balance, etc)
- Linked a second KYC'd Uphold wallet
- Verified  UI appears fine (UI, balance is shown), but logs show 403 error for linking `(https://grant.rewards.bravesoftware.com/v3/wallet/uphold/<paymentID>/claim)`

```
[ REQUEST ]
    > URL: https://grant.rewards.bravesoftware.com/v3/wallet/uphold/d3.......1/claim
    > Method: POST
    > Content: {"signedLinkingRequest":"eyJ......J9"}
    > Content Type: application/json; charset=utf-8
2021-08-31 06:40:34.738 19036-19036/? V/chromium: [VERBOSE6:logging_util.cc(136)] 
    [ RESPONSE - OnRequest ]
    > Url: https://grant.rewards.bravesoftware.com/v3/wallet/uphold/d32.......1/claim
    > Result: Failure
    > HTTP Code: 403
    > Body: {"message":"error linking wallet: unable to link wallets: unable to match wallets: wallets do not match","code":403}
```
- Quit Browser 
-  Upgraded to 1.29.x
- Verified 2nd verified, KYC'd wallet is no longer connected
- Verified UI shows pre-link state (UI, no balance)
```
    [ REQUEST ]
    > URL: https://grant.rewards.bravesoftware.com/v3/wallet/uphold/d3.....1/claim
    > Method: POST
    > Content: {"signedLinkingRequest":"ey......9"}
    > Content Type: application/json; charset=utf-8
2021-08-31 06:42:36.197 20591-20591/? V/chromium: [VERBOSE6:logging_util.cc(136)] 
    [ RESPONSE - OnRequest ]
    > Url: https://grant.rewards.bravesoftware.com/v3/wallet/uphold/d3.....1/claim
    > Result: Failure
    > HTTP Code: 403
    > Body: {"message":"error linking wallet: unable to link wallets: unable to match wallets: wallets do not match","code":403}
2021-08-31 06:42:36.205 20591-20591/? I/chromium: [INFO:post_claim_uphold.cc(123)] Forbidden
2021-08-31 06:42:36.210 20591-20591/? V/chromium: [VERBOSE1:uphold.cc(188)] Disconnecting wallet
2021-08-31 06:42:36.712 20591-20591/? V/chromium: [VERBOSE6:logging_util.cc(136)] 
    [ RESPONSE - OnRequest ]
    > Url: https://grant.rewards.bravesoftware.com/v3/wallet/uphold/d3......1/claim
    > Result: Failure
    > HTTP Code: 403
    > Body: {"message":"error linking wallet: unable to link wallets: unable to match wallets: wallets do not match","code":403}
2021-08-31 06:42:36.722 20591-20591/? I/chromium: [INFO:post_claim_uphold.cc(123)] Forbidden
```

`1.28.x Acc #1` | `1.28.x Acc #2` | 1.29.76
-- | -- | --
![image](https://user-images.githubusercontent.com/17010094/131489868-1326c119-2e83-4936-b350-727c22609826.jpg) | ![image](https://user-images.githubusercontent.com/17010094/131489880-eb10b7b7-3ea7-48ed-bd3c-b445c8e1bfab.jpg) | ![image](https://user-images.githubusercontent.com/17010094/131489889-d5fd1e88-3c1e-49a2-b569-0dd301efd911.jpg)


</details>

<details>
<summary>Scenario 5: <code>Upgrade (DISCONNECTED_NOT_VERIFIED) - pending->disconnected</code></summary>
<br/>

- Clean install on 1.28.x (current release)
- Enable Rewards on staging environment and close the browser
- Using Charles Proxy re-wrote Uphold user status to pending in `https://api-sandbox.uphold.com/v0/me` response
- Restart 1.28.x, connect Uphold account
- Verified PENDING on brave://rewards-internals and on wallet panel
- Go to uphold site, revoke access. Confirm DISCONNECTED_NOT_VERIFIED on brave://rewards-internals page.
- Close browser
- Upgrade to 1.29.76
- Should be in DISCONNECTED_VERIFIED state after upgrade (confirm on brave://rewards-internals)
- No notification shown due to #17780 

![image](https://user-images.githubusercontent.com/17010094/131504988-810a351b-813e-418c-8a9a-0cfbc9e235b6.png) | ![image](https://user-images.githubusercontent.com/17010094/131505068-0b9a0c6f-dbcc-4b32-b33f-ed9b4c17da85.png) | ![image](https://user-images.githubusercontent.com/17010094/131505172-078e4f78-beb0-4825-8e30-6879485fe9d7.png)
-- | -- | --

</details>

<details>
<summary>Scenario 6: <code>Upgrade (PENDING/blocked)</code></summary>
<br/>

- Clean install on 1.28.x (current release)
- Enable Rewards on staging environment and close the browser
- Using Charles Proxy re-wrote Uphold user status to pending in `https://api-sandbox.uphold.com/v0/me` response
- Restart 1.28.x, connect Uphold account
- Verified PENDING on brave://rewards-internals and on wallet panel
- Upgrade to 1.29.76
- Verified  brave://rewards-internals shows `PENDING ==> NOT_CONNECTED` in Event logs tab
- No notification shown due to #17780 

![image](https://user-images.githubusercontent.com/17010094/131505692-2007fb9a-0339-4f5d-9375-be599d8cf41e.png) | ![image](https://user-images.githubusercontent.com/17010094/131505455-4719e00e-dce5-42bd-93e0-b34369cd066e.png)
-- | --

</details>

<details>
<summary>Scenario 7: <code>New Notifications</code></summary>
<br/>
Not Implemented. Logged #17780
</details>

<details>
<summary>Scenario 8: <code>Clean profile - simulate natural Disconnect</code></summary>
<br/>

Scenario 1:

- Clean install on a non-JP region 
- Enable Rewards on staging environment
- Link a KYC Uphold wallet
- Verified wallet is linked (Wallet balance updates from Uphold)
- Visited Uphold account and revoked Brave-Browser access to simulate natural disconnect
- Verified opening BR Panel shows "Disconnected" UI 


Scenario 2:

- Clean install on a non-JP region 
- Enable Rewards on staging environment
- Link a KYC Uphold wallet
- Verified wallet is linked (Wallet balance updates from Uphold)
- Visited Uphold account and revoked Brave-Browser access to simulate natural disconnect
- Verified BR panel shows "Disconnected", no balance is displayed
- Verified brave://rewards-internals status is "Disconnected (Verified)"
- Verified able to tap on "Disconnected" on panel and be directed to re-authorize Brave Browser
- Verified able to re-auth successful
<img width=300 src=https://user-images.githubusercontent.com/17010094/131503463-7b77d057-ab1f-454b-a295-1aeaaaf6968b.png>

</details>

<details>
<summary>Scenario 9: <code>Semi-Verified state (Upgrade)</code></summary>
<br/>

Device limit has not been tested before on Android and since its a notification that is shown it may not be implemented so will have to wait for #17780 to be done to check this one

</details>
